### PR TITLE
Add Transaction Request API

### PR DESCRIPTION
**Pull request by adamaltman**

### CODE REVIEW

File: spec/components/schemas/Transactions/TransactionRequest.yaml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  type:
+    description: |
+      The type of transaction requested. 
+      You should always include the type within your API request.
+      This supports a limited subset of Transaction types.  To refund or void, use the refund endpoint.
+      To `capture` use the `sale` endpoint.  It will capture if there are any eligible
+      transactions of type `authorize` that can be captured.
+    type: string
+    enum:
+    - sale
```
Comment (ivizma) [inline spec/components/schemas/Transactions/TransactionRequest.yaml]: please fix alignment

Comment (adamaltman) [reply spec/components/schemas/Transactions/TransactionRequest.yaml]: I fixed it.  Thank you.

File: spec/components/schemas/Transactions/TransactionRequest.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+type: object
+required:
+  - websiteId
+  - customerId
+  - currency
+  - amount
+  - type
+properties:
```
Comment (dara123) [inline spec/components/schemas/Transactions/TransactionRequest.yaml]: should we have `redirectUrl` here too?

Comment (adamaltman) [reply spec/components/schemas/Transactions/TransactionRequest.yaml]: I don't think so... we can discuss it more.

**Comment by dara123** [comment]
need to merge this for https://github.com/Rebilly/backend/pull/3909 test

**Pull request merged**